src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx

```diff
@@ -0,0 +1,345 @@
+          }}
+        >
+          <FormGroup
+            isRequired
+            label={buildConfigEntityAttributes.name.title}
+            fieldId={buildConfigEntityAttributes.name.id}
+            helperText={
+              <FormHelperText isHidden={getFieldState(buildConfigEntityAttributes.name.id) !== 'error'} isError>
+                {getFieldErrors(buildConfigEntityAttributes.name.id)}
+              </FormHelperText>
+            }
+          >
+            <TextInput
+              isRequired
+              type="text"
+              id={buildConfigEntityAttributes.name.id}
+              name={buildConfigEntityAttributes.name.id}
+              autoComplete="off"
+              {...register<string>(buildConfigEntityAttributes.name.id, fieldConfigs.name)}
+            />
+          </FormGroup>
+
+          <FormGroup label={buildConfigEntityAttributes.description.title} fieldId={buildConfigEntityAttributes.description.id}>
+            <TextArea
+              id={buildConfigEntityAttributes.description.id}
+              name={buildConfigEntityAttributes.description.id}
+              autoResize
+              resizeOrientation="vertical"
+              {...register<string>(buildConfigEntityAttributes.description.id)}
+            />
+          </FormGroup>
+
+          <FormGroup
+            isRequired
+            label={buildConfigEntityAttributes.environment.title}
+            fieldId={buildConfigEntityAttributes.environment.id}
+            helperText={
+              <>
+                <FormHelperText isHidden={getFieldState(buildConfigEntityAttributes.environment.id) !== 'error'} isError>
+                  {getFieldErrors(buildConfigEntityAttributes.environment.id)}
+                </FormHelperText>
+
+                {selectedEnvironment?.deprecated && (
+                  <HelperText>
+                    <HelperTextItem hasIcon variant="warning">
+                      Selected Environment is deprecated
+                    </HelperTextItem>
+                  </HelperText>
+                )}
+              </>
+            }
+          >
+            <FormInput<string>
+              {...register<string>(buildConfigEntityAttributes.environment.id, fieldConfigs.environment)}
+              render={({ value, validated, onChange }) => (
+                <SearchSelect
+                  selectedItem={value}
+                  validated={validated}
+                  onSelect={(_, environment: Environment) => {
+                    onChange(environment.description!);
+                    setSelectedEnvironment(environment);
+                  }}
+                  onClear={() => {
+                    onChange('');
+                    setSelectedEnvironment(undefined);
+                  }}
+                  fetchCallback={fetchEnvironments}
+                  titleAttribute="description"
+                  placeholderText="Select Environment"
+                  getHighlightText={(environment: Environment) =>
```

Comment:
I think `getHighlightText` is too specific and there is high chance that there will be other requirements soon resulting in some other attributes which will be hard to maintain, to handle them more easily I would use general text modification instead, so that everyone can decide whether to use:
 - default text
 - custom text (if defined here)

Different use-cases may force you to use different "highlight" techniques, orders, etc. Using `customText` would be universal.

Example

```tsx
// ...
customText={(environment: Environemnt) => {
  if (environemt.deprecated) { ... }
  return <>`generating custom content here will cover also other possible use cases more easily`</>
}}
// ...
```